src/tcompiler.cpp

```diff
@@ -312,6 +312,10 @@ int terra_inittarget(lua_State *L) {
     else
         TT->CPU = llvm::sys::getHostCPUName().str();
 
+    if (TT->CPU == "generic") {
+        TT->CPU = "x86-64";
```

Comment:
I wonder if we can generalize this? There are systems we run on other than x86, so it would be nice not to hard code this quite as closely. E.g., perhaps we could consult `uname` or `/proc/cpuinfo` or similar (at least on Linux).

Reply:
This could be generalized, but this is mostly just a hack to get around LLVM being broken on certain linux installs. Any attempted generalization would have to eventually have a fallback to "x86-64".

Reply:
Help me understand the nature of this bug. Is it x86 specific? Could someone on ARM or PPC hit it, and if so what are they supposed to do about it if we use this patch?

Reply:
Erik's looked closer at the mechanics of the bug in LLVM than I have, so I can't say for sure whether it'll come up on ARM or PPC. The issue, as best I understand it, is that we'll have to either reimplement CPU detection ourselves, as a fallback, or we have to make an assumption about what `generic` means, as I've got in this pair of lines.

Reply:
Everyone can hit it, because LLVM simply returns "generic" [if anything fails](https://github.com/llvm/llvm-project/blob/d480f968ad8b56d3ee4a6b6df5532d485b0ad01e/llvm/lib/Support/Host.cpp#L1212) in direct conflict with the documentation, which is blatantly wrong. Technically any fix we put in here should actually be a patch to LLVM itself, but even trying to use `/proc/cpuinfo` may not always work for weird linux systems. I bring up "weird linux systems" because those are exactly the ones where LLVM's current detection will fail on.

Reply:
Ok, I guess this is fine for now. I agree doing our own CPU auto-detection is undesirable. If we had to do anything, I guess we could add a variable like `TERRA_DEFAULT_ARCH` or similar, but I don't know if this is worth it. We can always add it later if people actually run into this.